accord-core/src/main/java/accord/coordinate/tracking/FastPathTracker.java

```diff
@@ -30,17 +31,50 @@
 import static accord.coordinate.tracking.AbstractTracker.ShardOutcomes.*;
 
 // TODO (desired, efficiency): if any shard *cannot* take the fast path, and all shards have accepted, terminate
-public class FastPathTracker extends AbstractTracker<FastPathTracker.FastPathShardTracker, Node.Id>
+public class FastPathTracker extends AbstractTracker<FastPathTracker.FastPathQuorumShardTracker>
 {
     private static final ShardOutcome<FastPathTracker> NewFastPathSuccess = (tracker, shardIndex) -> {
         --tracker.waitingOnFastPathSuccess;
         return --tracker.waitingOnShards == 0 ? Success : NoChange;
     };
 
-    public static class FastPathShardTracker extends ShardTracker
+    public static class FastPathQuorumShardTracker extends QuorumShardTracker
     {
-        protected int fastPathAccepts, accepts;
-        protected int fastPathFailures, failures;
+        public FastPathQuorumShardTracker(Shard shard)
+        {
+            super(shard);
+        }
+
+        public ShardOutcome<? super FastPathTracker> onQuorumSuccess(Node.Id node)
+        {
+            return super.onSuccess(node);
+        }
+
+        public ShardOutcome<? super FastPathTracker> onMaybeFastPathSuccess(Node.Id node)
+        {
+            return super.onSuccess(node);
+        }
+
+        public ShardOutcome<? super FastPathTracker> onFailure(@Nonnull Node.Id from)
+        {
+            return super.onFailure(from);
+        }
+
+        public boolean hasRejectedFastPath()
+        {
+            return false;
+        }
+
+        public boolean hasMetFastPathCriteria()
+        {
+            return true;
```

Comment:
don't we want to disable fast path commits if we need to talk with multiple epochs, or has that been relaxed? I vaguely remember hearing that it was, but am not sure

Reply:
Only when talking to _new_ epochs; if the latest epoch is the same as the txnId then it's fine.